src/client/activation/migrateDataScienceSettingsService.ts

```diff
@@ -72,13 +79,38 @@ export class MigrateDataScienceSettingsService implements IExtensionActivationSe
         });
 
         await this.fs.writeLocalFile(filePath, fileContents);
-        return fileContents;
+    }
+
+    // Users may have mapped old python.dataScience.* commands to custom keybindings
+    // in their user keybindings.json. Ensure we migrate these too.
+    private async fixKeybindingsFile(filePath: string) {
+        const fileContents = await this.fs.readLocalFile(filePath);
+        const errors: ParseError[] = [];
+        const keybindings = parse(fileContents, errors, { allowTrailingComma: true, disallowComments: false });
+        if (errors.length > 0) {
+            traceError('JSONC parser returned ParseError codes', errors);
+            return;
+        }
+        if (!Array.isArray(keybindings)) {
+            return;
+        }
+        keybindings.forEach((keybinding) => {
+            if (typeof keybinding.command === 'string') {
+                keybinding.command = keybinding.command.replace('python.dataScience.', 'jupyter.');
```

Comment:
Same as preivous comment. 
We might end up creating duplicates.
Users may have created the new keyboard mappings & left the old ones.

Reply:
Couldn't find code that addresses this concern.
Please could you point me in that direction, I may have missed it